src/shared/Dapper/BaseDapperStorage.cs

```diff
@@ -2,11 +2,20 @@ namespace SharedContracts.Dapper;
 
 public abstract class BaseDapperStorage(ILogger<IDataStorage> logger, PostgresConfiguration configuration)
 {
+    private NpgsqlConnection CreateNpgsqlConnection()
+    {
+        var connectionStringBuilder = new NpgsqlConnectionStringBuilder(configuration.StorageConnectionString)
+        {
+            CommandTimeout = 3000,
```

Comment:
I wouldn't hard code the command timeout like this.
You can actually set this as part of your db connection string instead
https://www.npgsql.org/doc/connection-string-parameters.html

For example:
`Server=127.0.0.1;Port=5432;Database=myDataBase;User Id=myUsername;Password=myPassword;CommandTimeout=3000;`